lib/storage/partition.go

```diff
@@ -806,6 +809,11 @@ func (pt *partition) mustMergeInmemoryPartsFinal(pws []*partWrapper) *partWrappe
 	}
 	mpDst.ph = *ph
 
+	// resulting part is empty, no need to create a part wrapper
+	if ph.BlocksCount == 0 {
```

Comment:
Could we add test for this case somehow?

Reply:
Sure, added a test to cover this case